debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -328,25 +340,48 @@ private void initializeRedoLogsForMining(OracleConnection connection, boolean po
                 buildDataDictionary(connection);
             }
             if (!isContinuousMining) {
-                currentLogFiles = setLogFilesForMining(connection, startScn, archiveLogRetention, archiveLogOnlyMode,
-                        archiveDestinationName, logFileQueryMaxRetries, initialDelay, maxDelay);
-                currentRedoLogSequences = getCurrentLogFileSequences(currentLogFiles);
+                addLogFileForMining(connection, startScn);
             }
         }
         else {
             if (!isContinuousMining) {
                 if (OracleConnectorConfig.LogMiningStrategy.CATALOG_IN_REDO.equals(strategy)) {
                     buildDataDictionary(connection);
                 }
-                currentLogFiles = setLogFilesForMining(connection, startScn, archiveLogRetention, archiveLogOnlyMode,
-                        archiveDestinationName, logFileQueryMaxRetries, initialDelay, maxDelay);
-                currentRedoLogSequences = getCurrentLogFileSequences(currentLogFiles);
+                addLogFileForMining(connection, startScn);
             }
         }
 
         updateRedoLogMetrics();
     }
 
+    /**
+     *  add logfile for LogMiner
+     *
+     * @param connection database connection, should not be {@code null}
+     * @param startScn mining session's starting system change number (exclusive), should not be {@code null}
+     * @throws SQLException if a database exception occurred
+     */
+    private void addLogFileForMining(OracleConnection connection, Scn startScn) throws SQLException {
+        // If the time of startscn is 30 minutes later than the current time.
+        // The catch-up mode will be used. No need to load online redo log
+        final long onlineWindow = TimeUnit.MINUTES.toMillis(30);
+        LocalDateTime currentLocalTime = LocalDateTime.now();
+        Optional<LocalDateTime> startTime = connection.getScnToTimestamp(startScn).map(OffsetDateTime::toLocalDateTime);
```

Comment:
I'm a bit concerned with this usage here.

The default `undo_retention` database parameter is 900 seconds (15 minutes).  In my experience, I don't rely on users having changed many of these default values unless they've hit a situation where they need to, so on a basic installation of Oracle, you'll never hit a situation where this "catch-up" logic will fire.  This is because after the 15 minutes, the SCN has aged out of the Flashback Area and the `Optional` value will not be present.

I wonder if this "30" minute value should be defined as an internal configuration property.  This way we don't need to document the property but gives us a way to tune the behavior if someone's environment needs some adjustments due to the introduction of this feature.

Also, did you happen to test this scenario specifically using `log.mining.strategy=online_catalog` or did you also test this using the default mining strategy?  I'm specifically concerned when using the default strategy that if none of the archive logs being mined has the data dictionary, will Oracle LogMiner fail?

Reply:
Hi @elgca, just a gentle ping if you can answer my question above.

Reply:
I am using `log.mining.strategy=online_catalog` for testing，not fail. In my case, scn is 3 hours behind. It was running in catch up until I caught up with the scn in the last 30 minutes.

30 minutes is not a strict number, just the number of archivelogs in 30 minutes will not waste too much time in add_logfile, for me, so I tend to keep the original logic.

Reply:
@Naros

Reply:
![image](https://user-images.githubusercontent.com/22080060/170512093-a843e253-4572-4f38-8630-4e87af4a9ca9.png)

In fact, my archivelog interval is much less than 30min

Reply:
Hi @elgca its not so much about the archive log interval rate but what is your environment's undo retention.  You could check this in your environment by running this query:
```
SELECT name, value from v$parameter where name = 'undo_retention';
```